src/components/item.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledDeleteButton = styled.button`
+  display: none;
+  color: black;
+  :hover {
+    color: red;
+  }
+`;
+
+const StyledLi = styled.li`
+  display: flex;
+  justify-content: space-between;
+  padding: 0 var(--size-item-padding);
+  :hover ${StyledDeleteButton} {
+    display: initial;
+  }
+`;
+const StyledButtonContainer = styled.button`
+  width: 100%;
+`;
+
+const StyledDoneItem = styled.span`
+  text-decoration: 5px solid line-through;
+  color: var(--color-light-black);
+`;
+
+const Item = ({ content, state, updateItemList, itemList, id }) => {
+  // 현재 state를 입력받고, state를 갱신하여 업데이트
+  const toggleState = (previousState) => {
+    const nextState = previousState === 'todo' ? 'done' : 'todo';
```

Comment:
저는 state 자체를 true, false로 설정해 놓고
```suggestion
    const nextState = !previousState;
```
이렇게 구현했습니다!